src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -5638,4 +5653,34 @@ private Set<String> getGeneratedFeatures() {
                 generatedFeaturesFile, null, null);
         return genFeatSet;
     }
+
+    /**
+     * Create metadata when running devc mode and containers
+     * Language server then uses metadata file to connect
+     * 
+     */
+    public void writeDevcMetadata() {
+        File metaFile = new File(buildDirectory, serverDirectory.getName() + "-liberty-devc-metadata.xml");
+        try {
+            XMLStreamWriter metadataWriter = XMLOutputFactory.newInstance().createXMLStreamWriter(new FileWriter(metaFile)) ;
+            metadataWriter.writeStartDocument();
+            metadataWriter.writeStartElement("devModeMetaData");
+            writeElement(metadataWriter, "containerName", containerName != null ? containerName : DEVMODE_CONTAINER_BASE_NAME);
+            if (containerStartTimestamp != null) writeElement(metadataWriter, "startTime", containerStartTimestamp.toString());
```

Comment:
You changed these vars to Strings already. So no need to call `toString()` on them.